src/main.py

```diff
@@ -276,9 +288,13 @@ def main():
         except Exception as e:
             logging.error(e)
             time.sleep(10)
+        if killhandler.killed==True:
+            print ("Exited \"bot\" gracefully")
+            break
 
 
 if __name__ == "__main__":
+    killhandler=KillHandler()
```

Comment:
Another tiny nitpick but please add one space around assignments and comparison operator: so use  `foo = bar` instead of `foo=bar` and so forth